en/micro-integrator/docs/use-cases/examples/routing_examples/routing_based_on_payloads.md

```diff
@@ -0,0 +1,104 @@
+# Routing Based on Message Payloads
+    
+The example scenario uses two back-end services (IBM and MSFT) and illustrates how a request message is routed to the backend based on the message content (payload). If the request is made for IBM, the request is routed to the IBM stock inventory service. If the request is for MSFT, it is routed to the MSFT stock inventory service. 
+    
+## Synapse configuration
+    
+Listed below are the synapse configurations (proxy service) that are necessary for implementing this scenario. See the instructions on how to [build and run](#build-and-run) this example.
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<proxy name="ContentBasedRoutingProxy" xmlns="http://ws.apache.org/ns/synapse" transports="https http" startOnLoad="true" trace="disable">
+    <target>
+       <!-- When a request arrives the following sequence will be followed -->   
+       <inSequence>
+         <!-- The content of the incoming message will be isolated -->
+         <switch source="//ser:getQuote/ser:request/xsd:symbol" xmlns:ser="http://services.samples" xmlns:xsd="http://services.samples/xsd">
+                <case regex="IBM">
+                    <!-- Will Route the content to the appropriate destination -->  
+                    <send>
+                         <endpoint>
+                             <address uri="http://localhost:9000/services/SimpleStockQuoteService"/>
+                         </endpoint>
+                    </send>
+                </case>
+                <case regex="MSFT">
```

Comment:
We do the same thing in both cases (IBM,MSFT). Might not be the best way to express the idea. Shall we do something different in the case of 'MSFT'? May be create a custom payload and respond back saying 'invalid payload'? WDYT?

Reply:
@Vathsan is changing the backend to be run separately for MSFT and IBM. That should solve the problem right?

Reply:
Changed the synapse configuration to address this issue.